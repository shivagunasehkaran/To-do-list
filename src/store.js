// redux specific imports
import {applyMiddleware, createStore, compose} from 'redux';
import thunk from 'redux-thunk';
import rootReducer from './reducers/rootReducer';

let middleware = applyMiddleware(thunk);

const store = createStore(rootReducer, compose(middleware));

export default store;
