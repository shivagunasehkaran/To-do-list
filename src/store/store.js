// redux specific imports
import AsyncStorage from '@react-native-async-storage/async-storage';
import {applyMiddleware, createStore} from 'redux';
import {persistReducer, persistStore} from 'redux-persist';
import thunk from 'redux-thunk';
import rootReducer from '../reducers/rootReducer';

// Middleware : Redux Persist Config
const persistConfig = {
  key: 'root',
  storage: AsyncStorage,
  //Save specific Reducers
  whitelist: ['todo'],
  // Don't save specific Reducers
  blacklist: [],
};

// Middleware: Redux Persist Persisted Reducer
const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = createStore(persistedReducer, applyMiddleware(thunk));

// Middleware: Redux Persist Persister
let persistor = persistStore(store);

export {store, persistor};
