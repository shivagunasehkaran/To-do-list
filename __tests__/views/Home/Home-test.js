import enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import {mockStore} from '../../../src/store/__mocks__/store';
import ConnectedHome from '../../../src/views/Home/Home';

enzyme.configure({adapter: new Adapter()});

const defaultStore = {
  network: {isConnected: true},
};

describe('Home screen: Snapshot Testing', () => {
  it('renders correctly when success callback with all the required params in response', async () => {
    const data = {
      title: 'shiva',
      description: 'test',
    };
    const store = mockStore({
      ...defaultStore,
      todo: {
        todoDetails: {
          title: 'test',
          description: 'test',
        },
      },
      todoData: {data: data},
    });

    const tree = renderer
      .create(
        <Provider store={store}>
          <ConnectedHome />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
