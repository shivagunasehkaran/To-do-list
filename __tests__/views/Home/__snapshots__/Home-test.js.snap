// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home screen: Snapshot Testing renders correctly when success callback with all the required params in response 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F3F3F3",
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "marginBottom": 20,
          }
        }
      >
        <RCTScrollView
          contentContainerStyle={
            Object {
              "paddingBottom": 50,
            }
          }
          data={
            Object {
              "description": "test",
              "title": "test",
            }
          }
          disableVirtualization={false}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          onContentSizeChange={[Function]}
          onEndReachedThreshold={2}
          onLayout={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          updateCellsBatchingPeriod={50}
          viewabilityConfigCallbackPairs={Array []}
          windowSize={21}
        >
          <View />
        </RCTScrollView>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#8F8FFF",
            "bottom": 0,
            "height": 70,
            "justifyContent": "center",
            "opacity": 1,
            "position": "absolute",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Avenir-Medium",
              "fontSize": 20,
            }
          }
        >
          ADD
        </Text>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
